test/test_indexing.py

```diff
@@ -1558,6 +1558,31 @@ def test_take_along_dim_invalid(self, device, dtype):
         with self.assertRaisesRegex(IndexError, "Dimension out of range"):
             torch.take_along_dim(t, indices, dim=7)
 
+    @dtypes(torch.float)
+    def test_put_along_dim(self, device, dtype):
+        def _test_against_numpy(t, indices, values, dim):
+            actual = torch.put_along_dim(t, indices, values, dim)
+            t_np = t.cpu().numpy()
+            indices_np = indices.cpu().numpy()
+            values_np = values.cpu().numpy()
+            # performs operation in-place
+            np.put_along_axis(t_np, indices_np, values_np, dim)
+            self.assertEqual(actual, t_np, atol=0, rtol=0)
+
+        t = torch.tensor([[10, 30, 20], [60, 40, 50]], dtype=dtype)
```

Comment:
can you test the broadcasting behaviour between the different tensors, for both `dim` specified and `=None`?